Mlem/App/Views/Shared/Search/SearchView+FiltersView.swift

```diff
@@ -0,0 +1,217 @@
+//
+//  SearchView+FiltersView.swift
+//  Mlem
+//
+//  Created by Sjmarf on 08/09/2024.
+//
+
+import MlemMiddleware
+import SwiftUI
+
+extension SearchView {
+    @ViewBuilder
+    var filtersView: some View {
+        VStack(alignment: .leading, spacing: 0) {
+            ScrollView(.horizontal) {
+                HStack {
+                    switch selectedTab {
+                    case .communities:
+                        communityFiltersView
+                    case .users:
+                        personFiltersView
+                    case .instances:
+                        instanceFiltersView
+                    }
+                }
+                .padding(.vertical, 12)
+                .padding(.horizontal, Constants.main.standardSpacing)
+            }
+            .scrollIndicators(.hidden)
+            if selectedTab == .communities, communityFilters.instance.isOther {
+                Label(
+                    "Subscription statuses can't be displayed when using these filters.",
+                    systemImage: Icons.warning
+                )
```

Comment:
We can actually do this pretty easily by checking whether `(appState.firstSession as? UserSession)?.subscriptions.communities` contains a community with matching actor id--there's the slight challenge of tier, since that set expects `Community2`, but that should be fairly easily worked around to get the nice constant-time lookup behavior.

Reply:
Yeah, I did consider that. That fixes showing subscription status, but doesn't allow you to subscribe to or unsubscribe from the community. So, we'd still have to show some sort of warning to explain the absence of the "subscribe" action. There's probably some way to fix both at once, but I'm not quite sure exactly what it is yet and I didn't want to dedicate too much time to a relatively insignificant feature. Are you happy to skip it for now?

The answer might be to go full MVVM and implement a `CommunityViewModel` that stores a `Community` but remains fully interactable when that community's api is guest or inactive. If it is interacted with in that state, it would fetch and replace it's own `Community` with a new one from the active api, then perform the call on that new model. If replicated for posts, this system would also allow you to upvote/downvote/save posts in the feed after switching to a new account. There's probably a whole load of headaches associated with that approach though that I don't want to get into just yet - state faking, for one.

Reply:
Sure, that's fine--I'd rather avoid MVVM if possible, since SwiftUI is built to avoid it, so I'm content leaving OOS.